crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -34,7 +34,13 @@ pub(crate) struct AstIds {
 
 impl AstIds {
     fn expression_id(&self, key: impl Into<ExpressionNodeKey>) -> ScopedExpressionId {
-        self.expressions_map[&key.into()]
+        let key = &key.into();
+        match self.expressions_map.get(key) {
+            Some(result) => *result,
+            None => {
+                panic!("Could not find expression ID for {key:?}");
+            }
+        }
```

Comment:
I like this. You could do

```suggestion
        let key = ;
        match self.expressions_map.get(&key.into()).unwrap_or_else(|| {
	        panic!("Could not find expression ID for {key:?}");
        }
```

Reply:
Went with this, thanks for the suggestion!

Reply:
Great, went with that